Add switchbot cloud webhook and vaccum

Gigatrappeur opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add vacuum k10+ in switchbot_cloud integration.  
Add support of webhook for switchbot cloud

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

joostlek commented on homeassistant/components/switchbot_cloud/__init__.py:
```diff
@@ -109,6 +129,50 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
+        if not (persistent_config := await store.async_load()):
+            # Create config
+            persistent_config = {
+                CONF_WEBHOOK_ID: secrets.token_hex(),
```
Why are we storing this in a store instead of the config entry data?

Gigatrappeur replied:
Hum, I don't known.  
I'm new in ha development (but I'm a experienced developer in other side).  
So, I'm inspired in other integration like samsung smartthing but if there is a best way, lest's go

joostlek replied:
Withings is a good one to check

Gigatrappeur replied:
ok, I'll check it that  
many thanks

joostlek reviewed (commented):
Is there a way we can split off the webhook from the vacuum?

Gigatrappeur commented:
> Is there a way we can split off the webhook from the vacuum?

Yes, maybe. But, for my side, I have only vacuum k10+ and I need webhook for get status updated if I use the official application  
So, we can separate but webhook is necessary (I think) for vacuum integration  
  

On the other hand, if is exist a best solution, I'd be interested

joostlek commented:
In that case we have to combine it

jeeftor committed:
Bump Intellifire to 4.1.9 (#121091)

* rebase

* Minor patch to fix duplicate DeviceInfo beign created - if data hasnt updated yet

* rebase

* Minor patch to fix duplicate DeviceInfo beign created - if data hasnt updated yet

* fixing formatting

* Update homeassistant/components/intellifire/__init__.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update homeassistant/components/intellifire/__init__.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Removing cloud connectivity sensor - leaving local one in

* Renaming class to something more useful

* addressing pr

* Update homeassistant/components/intellifire/__init__.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* add ruff exception

* Fix test annotations

* remove access to private variable

* Bumping to 4.1.9 instead of 4.1.5

* A renaming

* rename

* Updated testing

* Update __init__.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* updateing styrings

* Update tests/components/intellifire/conftest.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Testing refactor - WIP

* everything is passing - cleanup still needed

* cleaning up comments

* update pr

* unrename

* Update homeassistant/components/intellifire/coordinator.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* fixing sentence

* fixed fixture and removed error codes

* reverted a bad change

* fixing strings.json

* revert renaming

* fix

* typing inother pr

* adding extra tests - one has a really dumb name

* using a real value

* added a migration in

* Update homeassistant/components/intellifire/config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update tests/components/intellifire/test_init.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* cleanup continues

* addressing pr

* switch back to debug

* Update tests/components/intellifire/conftest.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* some changes

* restore property mock cuase didnt work otherwise

* cleanup has begun

* removed extra text

* addressing pr stuff

* fixed reauth

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

mvn23 committed:
Split opentherm_gw entities between different devices (#124869)

* * Add migration from single device to multiple devices, removing all old entities
* Create new devices for Boiler and Thermostat
* Add classes for new entities based on the new devices

* Split binary_sensor entities into devices

* Split sensor entities into different devices

* Move climate entity to thermostat device

* Fix climate entity away mode

* Fix translation placeholders

* Allow sensor values with capital letters

* * Add EntityCategory
* Update and add device_classes

* Fix translation keys

* Fix climate entity category

* Update tests

* Handle `available` property in `entity.py`

* Improve GPIO state binary_sensor translations

* Fix: Updates are already subscribed to in the base entity

* Remove entity_id generation from sensor and binary_sensor entities

* * Use _attr_name on climate class instead of through entity_description
* Add type hints

* Rewrite to derive entities for all OpenTherm devices from a single base class

* Improve type annotations

* Use OpenThermDataSource to access status dict

* Move entity_category from entity_description to _attr_entity_category

* Move entity descriptions with the same translation_key closer together

* Update tests

* Add device migration test

* * Add missing sensors and binary_sensors back
* Improve migration, do not delete old entities from registry

* Add comments for migration period

* Use single lists for entity descriptions

* Avoid changing sensor values, remove translations

* * Import only required class from pyotgw
* Update tests

rikroe committed:
Fix BMW client blocking on load_default_certs (#125015)

* Fix BMW client blocking load_default_certs

* Use get_default_context

rikroe committed:
Fix telegram_bot blocking on load_default_certs (#125014)

* Fix telegram_bot blocking on load_default_certs

* Use sync variant of create_issue

mvn23 committed:
Migrate opentherm_gw climate entity unique_id (#125024)

* Migrate climate entity unique_id to match the format used by other opentherm_gw entities
* Add test to verify migration

rikroe committed:
Fix ollama blocking on load_default_certs (#125012)

* Fix ollama blocking on load_default_certs

* Use get_default_context instead of client_context

joostlek committed:
Bump aiomealie to 0.9.1 (#125017)

exxamalte committed:
Bump aio-georss-gdacs to 0.10 (#125021)

bump aio-georss-gdacs to 0.10

krasnoukhov committed:
Extend hjjcy device category in Tuya integration (#124854)

* Extend hjjcy device category in Tuya integration

* Better AQI level names

MartinHjelmare committed:
Bump python-telegram-bot to 21.5 (#125025)

EtienneSOU committed:
Fix Tado fan speed for AC (#122415)

* change capabilities

* fix tests 2

* improve usability with capabilities

* fix swings management

* Update homeassistant/components/tado/climate.py

Co-authored-by: Erwin Douna <e.douna@gmail.com>

* fix after Erwin's review

* fix after joostlek's review

* use constant

* use in instead of get

---------

Co-authored-by: Erwin Douna <e.douna@gmail.com>

Noltari committed:
Update aioairzone-cloud to v0.6.5 (#125030)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

dontinelli committed:
Add ConductivityConverter in websocket_api.py (#125029)

mib1185 committed:
Add diagnostics platform to modern forms (#125032)

bdraco committed:
Bump yarl to 1.9.7 (#125035)

thecode committed:
Bump aioshelly to 11.4.2 (#125036)

allenporter committed:
Clarify comment in google photos upload service (#125042)

gjohansson-ST committed:
New template merge_response (#114204)

* New template merge_response

* Extending

* Extend comment

* Update

* Fixes

* Fix comments

* Mods

* snapshots

* Fixes from discussion

emontnemery committed:
Fix motionblinds_ble tests (#125060)

tronikos committed:
Bump androidtvremote2 to 0.1.2 to fix blocking event loop when loading ssl certificate chain (#125061)

Bump androidtvremote2 to 0.1.2

epenet committed:
Improve renault tests (#125064)

dontinelli committed:
Bump fyta_cli to 0.6.6 (#125065)

bdraco committed:
Bump habluetooth to 3.4.0 (#125058)

changelog: https://github.com/Bluetooth-Devices/habluetooth/compare/v3.3.2...v3.4.0

Nidre committed:
Update Matter light transition blocklist to include YNDX LightStrip (#124657)

tronikos committed:
Add strict typing in Google Cloud (#125068)

tronikos committed:
Setup Google Cloud from the UI (#121502)

* Google Cloud can now be setup from the UI

* mypy

* Add BaseGoogleCloudProvider

* Allow clearing options in the UI

* Address feedback

* Don't translate Google Cloud title

* mypy

* Revert strict typing changes

* Address comments

emontnemery committed:
Remove unused event from recorder (#125067)

emontnemery committed:
Don't raise when registering entity service with invalid schema (#125057)

* Don't raise when registering entity service with invalid schema

* Update homeassistant/helpers/service.py

Co-authored-by: Robert Resch <robert@resch.dev>

---------

Co-authored-by: Robert Resch <robert@resch.dev>

LG-ThinQ-Integration committed:
Do not LG thinq retry entry setup, when a single coordinator failed (#125052)

Do not retry entry setup, when a single coordinator failed.

Co-authored-by: jangwon.lee <jangwon.lee@lge.com>

emontnemery committed:
Remove lying comment from service.async_register_entity_service (#125079)

emontnemery committed:
Deprecate template.attach (#124843)

sdb9696 committed:
Call async_write_ha_state after ring update (#125096)

Use async_write_ha_state after ring update

sdb9696 committed:
Bump ring-doorbell to 0.9.3 (#125087)

emontnemery committed:
Correct start version in recorder schema migration tests (#125090)

* Correct start version in recorder schema migration tests

* Remove default from states.last_updated_ts

dontinelli committed:
Remove roundig in Solarlog and add suggested_display_precision (#125094)

* Remove roundig and add suggested_display_precision

* Add suggested_unit_of_measurement

* Put lamda in parentheses

piitaya committed:
Update frontend to 20240902.0 (#125093)

dontinelli committed:
Add diagnostics to solarlog (#125072)

* Add diagnostics to solarlog

* Fix wrong comment

jbouwh committed:
Cleanup removed options for mqtt climate (#125083)

bieniu committed:
Add separate entities for temperature, humidity and pressure in AccuWeather integration (#125041)

* Add temperature, humidity and pressure sensors

* Make uv index sensor disabled by default

* Fix type

emontnemery committed:
Add test of statistics timestamp migration (#125100)

Djelibeybi committed:
Bump aiolifx to 1.0.9 and remove unused HomeKit model prefixes (#125055)

Co-authored-by: J. Nick Koston <nick@koston.org>

arturpragacz committed:
Fix area registry indexing when there is a name collision (#125050)

MartinHjelmare committed:
Handle telegram polling errors (#124327)

rikroe committed:
Fix blocking calls for OpenAI conversation (#125010)

bdraco committed:
Add coverage for late unifiprotect person detection events (#125103)

emontnemery committed:
Test string timestamps are wiped after migration to schema version 32 (#125091)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Bump yalexs to 8.6.0 (#125102)

Noltari committed:
Add Airzone Cloud Aidoo HVAC indoor/outdoor sensors (#125013)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

cnico committed:
Bump flipr api to 1.6.1 (#125106)

dontinelli committed:
Replace _host_in_configuration_exists with async_abort_entries_match in solarlog (#125099)

* Add diagnostics to solarlog

* Fix wrong comment

* Move to async_abort_entries_match

* Remove obsolete method solarlog_entries

* Update tests/components/solarlog/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update tests/components/solarlog/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update tests/components/solarlog/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update tests/components/solarlog/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Amend import of config_entries.SOURCE_USER

* Update tests/components/solarlog/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Ruff

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jbouwh committed:
Remove `is_on` function from `homeassistant.components` (#125104)

* Remove `is_on` method from `homeassistant.components`

* Cleanup test

emontnemery committed:
Remove recorder PostSchemaMigrationTask (#125076)

Co-authored-by: J. Nick Koston <nick@koston.org>

LG-ThinQ-Integration committed:
Add binary_sensor platform to LG Thinq (#125054)

* Add binary_sensor entity

* Update the document link due to the domain name change

* Update casing

---------

Co-authored-by: jangwon.lee <jangwon.lee@lge.com>

sdb9696 committed:
Convert ring integration to use entry.runtime_data (#125127)

arturpragacz committed:
Fix Onkyo action select_hdmi_output (#125115)

* Fix Onkyo service select_hdmi_output

* Move Hasskey directly under Onkyo domain

CFenner committed:
Correct device serial for ViCare integration (#125125)

* expose correct serial

* adapt inits

* adjust _build_entities

* adapt inits

* add serial data point

* update snapshot

* apply suggestions

* apply suggestions

dukeofphilberg committed:
Add Linkplay mTLS/HTTPS and improve logging (#124307)

* Work

* Implement 0.0.8 changes, fixup tests

* Cleanup

* Implement new playmodes, close clientsession upon ha close

* Implement new playmodes, close clientsession upon ha close

* Add test for zeroconf bridge failure

* Bump 0.0.9
Address old comments in 113940

* Exact _async_register_default_clientsession_shutdown

allenporter committed:
Add dependency on google-photos-library-api: Change the Google Photos client library to a new external package (#125040)

* Change the Google Photos client library to a new external package

* Remove mime type guessing

* Update tests to mock out the client library and iterators

* Update homeassistant/components/google_photos/media_source.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

iloveicedgreentea committed:
Bump py-madvr2 to 1.6.32 (#125049)

feat: update lib

sdb9696 committed:
Pass hass clientsession to ring config flow (#125119)

Pass hass clientsession to ring config flow

edenhaus committed:
Bump hadolint to 2.12.0 and use matrix for all Dockerfiles (#125131)

* Bump hadolint to 2.12.0 and use matrix for all Dockerfiles

* Fix

* Disable fail fast

bachya committed:
Fix unhandled exception with missing IQVIA data (#125114)

Michal4K committed:
Bump PySwitchbot to 0.48.2 (#125113)

emontnemery committed:
Log deprecation warning when `template.Template` is created without `hass` (#125142)

* Log deprecation warning when template.Template is created without hass

* Improve docstring

emontnemery committed:
Log deprecation warning when `cv.template` is called from wrong thread (#125141)

Log deprecation warning when cv.template is called from wrong thread

Bre77 committed:
Add climate platform to Tesla Fleet (#123169)

* Add climate

* docstring

* Add tests

* Fix limited scope situation

* Add another test

* Add icons

* Type vehicle data

* Replace inline temperatures

* Fix handle_vehicle_command type

* Fix preset turning HVAC off

* Fix cop_mode check

* Use constants

* Reference docs in command signing error

* Move to a read-only check

* Remove raise_for

* Fixes

* Tests

* Remove raise_for_signing

* Remove unused strings

* Fix async_set_temperature

* Correct tests

* Remove HVAC modes at startup in read-only mode

* Fix order of init actions to set hvac_modes correctly

* Fix no temp test

* Add handle command type

* Docstrings

* fix matches and fix a bug

* Split tests

* Fix issues from rebase

edenhaus committed:
Improve hassfest docker image (#125133)

* Improve hassfest docker image

* Use fixed uv version

* Use cli params instead env

* run hassfest

* Exclude pycache

sdb9696 committed:
Abort ring config_flow if account is already configured (#125120)

* Abort ring config_flow if account is already configured

* Update tests/components/ring/test_config_flow.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

tronikos committed:
Add Google Cloud Speech-to-Text (STT) (#120854)

* Google Cloud

* .

* fix

* mypy

* add tests

* Update .coveragerc

* Update const.py

* upload file, reconfigure and import flow

* fixes

* default to latest_short

* mypy

* update

* Allow clearing options in the UI

* update

* update

* update

marcelveldt committed:
Fix energy sensor for ThirdReality Matter powerplug (#125140)

emontnemery committed:
Restore unnecessary assignment of Template.hass in event helper (#125143)

arturpragacz committed:
Add Onkyo Receiver class to improve typing (#124190)

bdraco committed:
Speed up hassio send_command url check (#125122)

* Speed up hassio send_command url check

The send_command call checked the resulting path to make
sure that the input path was not modified when converting
to a URL. Since the host is is pre-set, we only need to check
the processed raw_path matches command instead of converting
back to a string, and than comparing it against another
constructed string.

* Speed up hassio send_command url check

The send_command call checked the resulting path to make
sure that the input path was not modified when converting
to a URL. Since the host is is pre-set, we only need to check
the processed raw_path matches command instead of converting
back to a string, and than comparing it against another
constructed string.

* adjust

emontnemery committed:
Add explaining comments in cv.template tests (#125081)

jschlyter committed:
Support setting Amazon Polly engine in service call (#120226)

dependabot[bot] committed:
Bump actions/upload-artifact from 4.3.6 to 4.4.0 (#125056)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4.3.6 to 4.4.0.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4.3.6...v4.4.0)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

s0129 committed:
Extended epson projector integration to include serial connections (#121630)

* Extended epson projector integration to include serial connections

* Fix review changes

* Improve epson types and translations

* Fix comment

---------

Co-authored-by: Joostlek <joostlek@outlook.com>

MartinHjelmare committed:
Clean up Z-wave error log when raising in service handlers (#125138)

gjohansson-ST committed:
End deprecation setting attributes directly on config entry (#123729)

* End deprecation setting attr directly on config entry

* Update ollama test

* Fix android_tv

mvn23 committed:
Remove opentherm_gw options migration (#125046)

UltimateGG committed:
Fix updating insteon modem configuration while disconnected (#121918)

#121917 Fix updating insteon modem configuration while disconnected

mib1185 committed:
Use SnapshotAssertion in more AVM Fritz!Box Tools tests (#125037)

use SnapshotAssertion in more tests

mjj4791 committed:
Increase timeout for fetching buienradar weather data (#124597)

Increase timeout for fetching weather data

rajlaud committed:
Add favorites support to Media Browser for Squeezebox integration (#124732)

* Add Favorites support to Media Browser

* CI fixes

* More CI Fixes

* Another CI

* Change icons for other library items to use standard LMS icons

* Change max favorites to BROWSE_LIMIT

* Simplify library_payload to consolidate favorite and non-favorite items

* Simplify library_payload to consolidate favorite and non-favorite items

* Add support for favorite hierarchy

* small fix for icon naming with local albums

* Add ability to expand an album from a favorite list

* Reformat to fix linting error

* and ruff format

* Use library calls from pysqueezebox

* Folder and playback support

* Bump to pysqueezebox 0.8.0

* Bump pysqueezebox version to 0.8.1

* Add unit tests

* Improve unit tests

* Refactor tests to use websockets and services.async_call

* Apply suggestions from code review

---------

Co-authored-by: peteS-UK <64092177+peteS-UK@users.noreply.github.com>

hanskroner committed:
Bump PyMetno to 0.13.0 (#125151)

emontnemery committed:
Make optional arguments to frame.report kwarg only (#125062)

* Make optional arguments to frame.report kwarg only

* Update homeassistant/helpers/frame.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

emontnemery committed:
Improve some comments in recorder tests (#125118)

andrew-codechimp committed:
Bump aiomealie to 0.9.2 (#125153)

Bump mealie version

ollo69 committed:
Add sensors for AsusWRT using http(s) library (#124337)

* Additional sensors for AsusWRT using http(s) library

* Remove temperature sensors refactor from PR

* Fix test function name

* Change translation a suggested

* Requested changes

alexandrecuer committed:
Migrate emoncms to config flow (#121336)

* Migrate emoncms to config flow

* tests coverage 98%

* use runtime_data

* Remove pyemoncms bump.

* Remove not needed yaml parameters add async_update_data to coordinator

* Reduce snapshot size

* Remove CONF_UNIT_OF_MEASUREMENT

* correct path in emoncms_client mock

* Remove init connexion check
as done by config_entry_first_refresh
since async_update_data catches exceptionand raise UpdateFailed

* Remove CONF_EXCLUDE_FEEDID from config flow

* Update homeassistant/components/emoncms/__init__.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/emoncms/sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/emoncms/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Use options in options flow and common strings

* Extend the ConfigEntry type

* Define the type explicitely

* Add data description in strings.json

* Update tests/components/emoncms/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update tests/components/emoncms/test_config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Add test import same yaml conf + corrections

* Add test user flow

* Use data_description...

* Use snapshot_platform in test_sensor

* Transfer all fixtures in conftest

* Add async_step_choose_feeds to ask flows to user

* Test abortion reason in test_flow_import_failure

* Add issue when value_template is i yaml conf

* make text more expressive in strings.json

* Add issue when no feed imported during migration.

* Update tests/components/emoncms/test_config_flow.py

* Update tests/components/emoncms/test_config_flow.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Alexwijn committed:
Add support for total YouTube views (#123144)

* Add support for retrieving the total views of a channel.

* Add missing tests

* Re-order imports

* Another run on code format

* Add missing translation

* Update YouTube test snapshots

rajlaud committed:
Enable strict typing for the Squeezebox integration (#125161)

* Strict typing for squeezebox

* Improve unit tests

* Refactor tests to use websockets and services.async_call

* Apply suggestions from code review

* Fix merge conflict

mvn23 committed:
Update opentherm_gw tests to avoid patching internals (#125152)

* Update tests to avoid patching internals

* * Use fixtures for tests
* Update variable names in tests for clarity

* Use hass.config_entries.async_setup instead of setup.async_setup_component

N3rdix committed:
Enhance error handling when changing a timer's duration (#121786)

* Update remaining before checking duration

* fix comment

* calculation based on transient field

* lint

* remove useless brackets

rajlaud committed:
Fix type errors in squeezebox (#125166)

piitaya committed:
Update frontend to 20240903.1 (#125160)

bdraco committed:
Bump yalexs to 8.6.2 (#125162)

changelog: https://github.com/bdraco/yalexs/compare/v8.6.0...v8.6.2

hanskroner committed:
Expose UV Index in Met.no (#124992)

UV Index now also appears in forecasts.

mvn23 committed:
Update opentherm_gw tests to prepare for new platforms (#125172)

Move MockConfigEntry to a fixture

elupus committed:
Update gardena_bluetooth dependency to 1.4.3 (#125175)

bdraco committed:
Bump yalexs to 8.6.3 (#125176)

Fixes the battery state not refreshing due to a refactoring
error in the library.

changelog: https://github.com/bdraco/yalexs/compare/v8.6.2...v8.6.3

ludeeus committed:
Add include-hidden-files to upload env_file artifact (#125179)

emontnemery committed:
Refactor recorder EventIDPostMigration data migrator (#125126)

bieniu committed:
Add re-auth flow to NextDNS integration (#125101)

gjohansson-ST committed:
Bump python-holidays to 0.56 (#125182)

starkillerOG committed:
Add 100% coverage of Reolink update platform (#124521)

* Add 100% update test coverage

* Add assertion

emontnemery committed:
Improve recorder tests to check indices are removed (#125164)

thecode committed:
Add myself as codeowner for BTHome (#125184)

ldfandian committed:
Bump xiaomi-ble to 0.31.1 to add support for human presence sensor XMOSB01XS (#124751)

raman325 committed:
Get zwave_js statistics data from model (#120281)

* Get zwave_js statistics data from model

* Add migration logic

* Update comment

* revert change to forward entry

emontnemery committed:
Fix missing patch in nextdns tests (#125195)

emontnemery committed:
Move recorder EntityIDPostMigrationTask to migration (#125136)

* Move recorder EntityIDPostMigrationTask to migration

* Update test

emontnemery committed:
Teach recorder data migrator base class to remove index (#125168)

* Teach recorder data migrator base class to remove index

* Fix tests

farmio committed:
Update knx-frontend to 2024.9.4.64538 (#125196)

bramkragten committed:
Update frontend to 20240904.0 (#125206)

bdraco committed:
Bump yarl to 1.9.8 (#125193)

changelog: https://github.com/aio-libs/yarl/compare/v1.9.7...v1.9.8

edenhaus committed:
Bump deebot-client to 8.4.0 (#125207)

LennP committed:
Add Motionblinds Bluetooth diagnostics (#121899)

* Add diagnostics platform

* Add diagnostics test

* Remove comments

* Exclude created_at and modified_at from snapshot

* Fix entry_id in mock_config_entry

* Add repr to excluded props from snapshot

* Improve diagnostics

* Use function name instead of number for callback diagnostics

* Remove info from diagnostics

* Reformat

starkillerOG committed:
Add 100% coverage of Reolink button platform (#124380)

* Add 100% button coverage

* review comments

* fix

* Use SERVICE_PRESS constant

* Use DOMAIN instead of const.DOMAIN

* styling

* User entity_registry_enabled_by_default fixture

* fixes

* Split out ptz_move test

* use SERVICE_PTZ_MOVE constant

hesselonline committed:
Add new number component for setting the wallbox ICP current (#125209)

* Add new number component for setting the wallbox ICP current

* feat: Add number component for wallbox ICP current control

bdraco committed:
Bump aioesphomeapi to 25.3.2 (#125188)

changelog: https://github.com/esphome/aioesphomeapi/compare/v25.3.1...v25.3.2

Michal4K committed:
Allow Switchbot users to force nightlatch (#124326)

* Add option to force nightlatch operation mode

* Fix format

* Make the new option available only for lock pro entry

* use senor_type instead of switchbot model + tests

* Update homeassistant/components/switchbot/lock.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

cdce8p committed:
Fix enum lookup (#125220)

edenhaus committed:
Update modified_at datetime on storage collection changes (#125218)

Shulyaka committed:
Update Anthropic default model to Haiku (#125225)

iskrakranj committed:
Add Iskra integration (#121488)

* Add iskra integration

* iskra non resettable counters naming fix

* added iskra config_flow test

* fixed iskra integration according to code review

* changed iskra config flow test

* iskra integration, fixed codeowners

* Removed counters code & minor fixes

* added comment

* Update homeassistant/components/iskra/__init__.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Updated Iskra integration according to review

* Update homeassistant/components/iskra/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Updated iskra integration according to review

* minor iskra integration change

* iskra integration changes according to review

* iskra integration changes according to review

* Changed iskra integration according to review

* added iskra config_flow range validation

* Fixed tests for iskra integration

* Update homeassistant/components/iskra/coordinator.py

* Update homeassistant/components/iskra/config_flow.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Fixed iskra integration according to review

* Changed voluptuous schema for iskra integration and added data_descriptions

* Iskra integration tests lint error fix

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

LG-ThinQ-Integration committed:
Bump thinqconnect to 0.9.6 (#125155)

* Refactor LG ThinQ integration

* Rename ha_bridge_list to bridge_list

* Update for reviews

* Correct spells
Do not use mqtt related api

* Guarantee update status

* Update for reviews

* Update reviews

---------

Co-authored-by: jangwon.lee <jangwon.lee@lge.com>

cdce8p committed:
Fix Path.__enter__ DeprecationWarning in tests (#125227)

synesthesiam committed:
Bump intents to 2024.9.4 (#125232)

thecode committed:
Fix BTHome validate triggers for device with multiple buttons (#125183)

* Fix BTHome validate triggers for device with multiple buttons

* Remove None default

martinssipenko committed:
Reenable Smarty integration (#124148)

* Reenable Smarty integration

* Updated codeowners to myself

* Revert "Updated codeowners to myself"

This reverts commit 639fef32b90d22117938f864e6ea3c55b0fc5074.

* Upgraded pysmarty2 to version 0.10.1 which is not pinned to specific pymodbus version

* Update requirements_all.txt

DCSBL committed:
Remove ExternalDevice migration in HomeWizard (#125197)

epenet committed:
Improve config flow type hints in starline (#125202)

epenet committed:
Improve config flow type hints (q-s) (#125198)

* Improve config flow type hints (q-s)

* Revert screenlogic

* Revert starline

CFenner committed:
Fix device identifier in ViCare integration (#124483)

* use correct serial

* add migration handler

* adjust init call

* add missing types

* adjust init call

* adjust init call

* adjust init call

* adjust init call

* Update types.py

* fix loop

* fix loop

* fix parameter order

* align parameter naming

* remove comment

* correct init

* update

* Update types.py

* correct merge

* revert type change

* add test case

* add helper

* add test case

* update snapshot

* add snapshot

* add device.serial data point

* fix device unique id

* update snapshot

* add comments

* update nmigration

* fix missing parameter

* move static parameters

* fix circuit access

* update device.serial

* update snapshots

* remove test case

* Update binary_sensor.py

* convert climate entity

* Update entity.py

* update snapshot

* use snake case

* add migration test

* enhance test case

* add test case

* Apply suggestions from code review

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

epenet committed:
Improve config flow type hints in screenlogic (#125199)

tl-sl committed:
Add Button platform for Smlight integration (#124970)

* Add button platform for smlight integration

* Add strings required for button platform

* Add commands api to smlight mock client

* Add tests for smlight button platform

* Move entity category to class

* Disable by default Zigbee flash mode

arturpragacz committed:
Add Onkyo to strict typing (#124617)

mvn23 committed:
Add button platform to opentherm_gw (#125185)

* Add button platform to opentherm_gw

* Add tests for button.py

* Update tests/components/opentherm_gw/test_button.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

synesthesiam committed:
Add support for sample bytes in preferred TTS format (#125235)

PeteRager committed:
Add Sonos tests and update error handling for unknown media (#124578)

* initial commit

* simplify tests

bdraco committed:
Revert "Disable IPv6 in the opower integration to fix AEP utilities" (#125208)

Revert "Disable IPv6 in the opower integration to fix AEP utilities (#107203)"

This reverts commit 2a9a046fab2ff3cde2ede62a50c253d5454b62de.

gjohansson-ST committed:
Remove deprecated aux_heat from honeywell (#125248)

taltaub22 committed:
Add Custom Drink Entities Tami4 Edge (#124506)

* Add drinks as button entities instead of using actions

* Remove button extensions

* Add an extension to create new buttons

* Use translation key for buttons names

* Change translation key wording

* Call async_add_entities once

* Add icons

* Update homeassistant/components/tami4/button.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

tronikos committed:
Add debug logging in get_cost_reads in opower (#124473)

Add debug statements in get_cost_reads in opower

tl-sl committed:
Add uptime sensors for Smlight (#124408)

* Add uptime sensor as derived sensor class

* Add strings for uptime sensors

* Update sensor tests to include uptime sensors

* test zigbee uptime when disconnected

CFenner committed:
Add hysteresis entity for heat pumps via ViCare (#124294)

* add hysteresis entity

* update PyViCare-neo dependency

* add hysteresis switch on / of entities

* Revert "add hysteresis entity"

This reverts commit dcb5680d0ca1958640e68de36f6befbf6416ab41.

gjohansson-ST committed:
Fix blocking call in yale_smart_alarm (#125255)

gjohansson-ST committed:
Don't show input panel if default code provided in envisalink (#125256)

shaiu committed:
Add service to 17track to archive package (#123493)

* Add service archive package

* Update homeassistant/components/seventeentrack/icons.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* CR fix in tests

* CR fix in services.py

* string references

* extract constant keys

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

iloveicedgreentea committed:
Add madvr diagnostics (#125109)

* feat: add basic diagnostics

* fix: add mock data

* fix: regen snapshots

thecode committed:
Address BTHome review comment (#125259)

* Address BTHome review comment

* Review comment

Co-authored-by: Ernst Klamer <e.klamer@gmail.com>

* generator expression

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

---------

Co-authored-by: Ernst Klamer <e.klamer@gmail.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

rajlaud committed:
Squeezebox remove deprecated sync and unsync services (#125271)

* Squeezebox remove deprecated sync and unsync

* Squeezebox remove sync group attribute

Jordi1990 committed:
Increase AquaCell timeout and handle timeout exception properly (#125263)

* Increase timeout and add handling of timeout exception

* Raise update failed instead of config entry error

bdraco committed:
Fix yarl binary wheel builds for armv7l and armhf (#125270)

bdraco committed:
Bump yarl to 1.9.9 (#125264)

cdce8p committed:
Use TypeVar defaults for Generator (#125228)

bdraco committed:
Switch hassio to use with_path where possible (#125268)

* Switch hassio to use with_path where possible

Any place we are joining to the root url, we can use with_path
as its much faster

* revert

silamon committed:
Add follower to the PlayingMode enum  (#125294)

Update media_player.py

emontnemery committed:
Teach recorder data migrator base class to update MigrationChanges (#125214)

* Teach recorder data migrator base class to update MigrationChanges

* Bump migration version

* Improve test coverage

* Update migration.py

* Revert migrator version bump

* Remove unneeded change

epenet committed:
Improve config flow type hints in ukraine_alarm (#125302)

emontnemery committed:
Simplify generic decorators in recorder (#125301)

* Simplify generic decorators in recorder

* Remove additional case

exxamalte committed:
Add diagnostics to GDACS integration (#125296)

* simple diagnostics

* add service status information

* remove from no diagnostics list

* wip

* cater for the case where status info is undefined

* make test work

* code reformatted

* add snapshot data

* simplify code

tl-sl committed:
Add reauth flow for Smlight (#124418)

* Add reauth flow for smlight integration

* add strings for reauth

* trigger reauth flow on authentication errors

* Add tests for reauth flow

* test for update failed on auth error

* restore name title placeholder

* raise config entry error to trigger reauth

* Add test for reauth triggered at startup

---------

Co-authored-by: Tim Lunn <tim@feathertop.org>

exxamalte committed:
Add diagnostics to GeoNet NZ Quakes integration (#125320)

* add diagnostics platform

* add tests

* add snapshot data

* remove from no diagnostics list

mrpasztoradam committed:
Add new data types to ADS integration (#125201)

* feat: Introduce new data types to ADS integration.

* refactor: ADS data unpacking based on PLC data type

* refactor: handle BOOL and STRING as special cases.

mvn23 committed:
Split opentherm_gw entity base class (#125330)

Add OpenThermStatusEntity to allow entities that don't need status updates

emontnemery committed:
Refactor recorder retryable_database_job decorator (#125306)

pssc committed:
Add Squeezebox server service binary sensors (#122473)

* squeezebox add binary sensor + coordinator

* squeezebox add connected via for media_player

* squeezebox add Player type for player

* Add more type info

* Fix linter errors

* squeezebox use our own status entity

* squeezebox rework device handling based on freedback

* Fix device creation

* squeezebox rework coordinator error handling

* Fix lint type error

* Correct spelling

* Correct spelling

* remove large comments

* insert small comment

* add translation support

* Simply sensor

* clean update function, minimise comments to the useful bits

* Fix after testing

* Update homeassistant/components/squeezebox/entity.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* move data prep out of Device assign for clarity

* stop being a generic api

* Humans need to read the sensors...

* ruff format

* Humans need to read the sensors...

* Revert "ruff format"

This reverts commit 8fcb8143e7c4427e75d31f9dd57f6c2027f8df6a.

* ruff format

* Humans need to read the sensors...

* errors after testing

* infered

* drop context

* cutdown coordinator for the binary sensors

* add tests for binary sensors

* Fix import

* add some basic media_player tests

* Fix spelling and file headers

* Fix spelling

* remove uuid and use service device cat

* use diag device

* assert execpted value

* ruff format

* Update homeassistant/components/squeezebox/__init__.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Simplify T/F

* Fix file header

* remove redudant check

* remove player tests from this commit

* Fix formatting

* remove unused

* Fix function Type

* Fix Any to bool

* Fix browser tests

* Patch our squeebox componemt not the server in the lib

* ruff

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

peteS-UK committed:
Fix typo in squeezebox (#125352)

Spelling Correction on SERVER_MODEL

bouwew committed:
Bump plugwise to v1.0.0 (#125354)

beastie29a committed:
Home Connect add FridgeFreezer switch entities (#122881)

* Home Connect add FridgeFreezer switch entities

* Fix unrelated test

* Implemented requested changes from review

* Move exist_fn check code to setup

* Assign entity_description during init

* Resolve issue with functional testing

YogevBokobza committed:
Bump aioswitcher to 4.0.3 (#125355)

gjohansson-ST committed:
Remove deprecated aux_heat from zha (#125247)

Remove aux_heat from zha

markruys committed:
Add Sensoterra integration (#119642)

* Initial version

* Baseline release

* Refactor based on first PR feedback

* Refactoring based on second PR feedback

* Initial version

* Baseline release

* Refactor based on first PR feedback

* Refactoring based on second PR feedback

* Refactoring based on PR feedback

* Refactoring based on PR feedback

* Remove extra attribute soil type

Soil type isn't really a sensor, but more like a configuration entity.
Move soil type to a different PR to keep this PR simpler.

* Refactor SensoterraSensor to a named tuple

* Implement feedback on PR

* Remove .coveragerc

* Add async_set_unique_id to config flow

* Small fix based on feedback

* Add test form unique_id

* Fix

* Fix

---------

Co-authored-by: Joostlek <joostlek@outlook.com>

gjohansson-ST committed:
Extend deprecation of aux_heat in ClimateEntity (#125360)

balloob committed:
Add model ID to Sonos (#125364)

balloob committed:
Add model ID to roborock (#125366)

balloob committed:
Add model ID to samsungtv (#125369)

balloob committed:
Add model ID to Wemo (#125368)

balloob committed:
Add model ID to awair (#125373)

* Add model ID to awair

* less diff

balloob committed:
Add model ID to unifiprotect (#125376)

synesthesiam committed:
Add assist satellite entity component (#125351)

* Add assist_satellite

* Update homeassistant/components/assist_satellite/manifest.json

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

* Update homeassistant/components/assist_satellite/manifest.json

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

* Add platform constant

* Update Dockerfile

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Address comments

* Update docstring async_internal_announce

* Update CODEOWNERS

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

emontnemery committed:
Correct check for removed index in recorder test (#125323)

emontnemery committed:
Don't allow templating min, max, step in config entry template number (#125342)

autinerd committed:
Bump ruff to 0.6.4 (#125385)

* Bump ruff to 0.6.4

* fix Dockerfile

tl-sl committed:
Add binary_sensor platform for Smlight integration (#125284)

* Support binary_sensors for SMLight integration

* Add strings for binary sensors

* Add tests for binary_sensor platform

* Update binary sensor docstring

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Regenerate snapshot

---------

Co-authored-by: Shay Levy <levyshay1@gmail.com>
Co-authored-by: Tim Lunn <tim@feathertop.org>

alengwenus committed:
Bump pypck to 0.7.22 (#125389)

danzel committed:
Fix controlling AC temperature in airtouch5 (#125394)

Fix controlling AC temperature

alengwenus committed:
Add tests for LCN actions / services (#125391)

* Add tests for services/actions

* Add snapshots for services/actions

* Use constants for service names and parameters

* Remove snapshot names

balloob committed:
Add model ID to linkplay (#125370)

tl-sl committed:
Bump pysmlight to 0.0.14 (#125387)

Bump pysmlight 0.0.14 for smlight

dontinelli committed:
Increase coordinator update_interval for fyta (#125393)

* Increase update_interval

* Update homeassistant/components/fyta/coordinator.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

autinerd committed:
Enable Ruff PTH for the script directory (#124441)

* Enable Ruff PTH for the script directory

* Address review comments

* Fix translations script

* Update script/hassfest/config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

epenet committed:
Replace SW version by model ID in renault device info (#125399)

* Replace SW_VERSION by MODEL_ID in renault device info

* Simplify PR

* Fix tests

Commit:
Lyric: fixed missed snake case conversions (#125382)

fixed missed snake case conversions

thecode committed:
Use smlight discovery hostname as device name (#125359)

* Use smlight discovery hostname as device name

* Update reauth flow name

* Drop host from description

jesperraemaekers committed:
Add weheat core integration (#123057)

* Add empty weheat integration

* Add first sensor to weheat integration

* Add weheat entity to provide device information

* Fixed automatic selection for a single heat pump

* Replaced integration specific package and removed status sensor

* Update const.py

* Add reauthentication support for weheat integration

* Add test cases for the config flow of the weheat integration

* Changed API and OATH url to weheat production environment

* Add empty weheat integration

* Add first sensor to weheat integration

* Add weheat entity to provide device information

* Fixed automatic selection for a single heat pump

* Replaced integration specific package and removed status sensor

* Add reauthentication support for weheat integration

* Update const.py

* Add test cases for the config flow of the weheat integration

* Changed API and OATH url to weheat production environment

* Resolved merge conflict after adding weheat package

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Added translation keys, more type info and version bump the weheat package

* Adding native property value for weheat sensor

* Removed reauth, added weheat sensor description and changed discovery of heat pumps

* Added unique ID of user to entity

* Replaced string by constants, added test case for duplicate unique id

* Removed duplicate constant

* Added offline scope

* Removed re-auth related code

* Simplified oath implementation

* Cleanup tests for weheat integration

* Added oath scope to tests

---------

Co-authored-by: kjell-van-straaten <kjell.van.straaten@wefabricate.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

balloob committed:
Small Assist Satellite fixes (#125384)

marcelveldt committed:
Fix for Hue sending effect None at turn_on command while no effect is active (#125377)

* Fix for Hue sending effect None at turn_on command while no effect is active

* typo

* update tests

tl-sl committed:
Add Switch platform to Smlight integration (#125292)

* Add switch platform to Smlight

* Add strings for switch platform

* Add tests for Smlight switch platform

* Regenerate snapshot

* Address review comments

* Use is_on property for updating switch state

* Address review comments

---------

Co-authored-by: Tim Lunn <tim@feathertop.org>

matrixd2 committed:
Add YoLink temperature sensor YS8008 support (#125408)

Add YS8008 support

ms264556 committed:
Rename "Ruckus Unleashed" integration to "Ruckus" (#125392)

steffenrapp committed:
Remove attributes from Nuki entities (#125348)

* Remove attributes from Nuki entities

* remove tests

piitaya committed:
Update frontend to 20240906.0 (#125409)

epenet committed:
Improve config flow type hints in volumio (#125318)

jrieger committed:
Add translations to Xiaomi Miio (#123822)

* Add translations to Xiaomi Miio

* Deduplicate translations

silamon committed:
Improve play media support in LinkPlay (#125205)

Improve play media support in linkplay

epenet committed:
Improve config flow type hints (t-z) (#125315)

shedokan committed:
Add Habitica up/down attributes for tasks (#125356)

add: up/down

epenet committed:
Improve config flow type hints in xiaomi_aqara (#125316)

epenet committed:
Improve config flow type hints in vulcan (#125308)

* Improve config flow type hints in vulcan

* Adjust tests

GeoffAtHome committed:
Improve geniushub test coverage (#124157)

* Add tests for local connection

* Test cloud setup

* Improve tests.

* Simplied coverage test to cloud setup.

* Mock out library and add snapshots

* Mock out library and add snapshots

* Update tests/components/geniushub/conftest.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Attempt to make it nice

* Fix

---------

Co-authored-by: Joostlek <joostlek@outlook.com>
Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

fiste788 committed:
Add binary sensors for motion detection Comelit simple home (#125200)

* Add binary sensors for motion detection

* sort platforms

* use _attr_device_class property and optimizations

* use static _attr_device_class property

epenet committed:
Fix location_id datatype in totalconnect tests (#125298)

Adjust location_id type in totalconnect tests

epenet committed:
Improve config flow type hints in totalconnect (#125300)

epenet committed:
Improve config flow type hints in tellduslive (#125299)

noahhusby committed:
Bump aiorussound to 3.0.4 (#125285)

feat: bump aiorussound to 3.0.4

bdraco committed:
Bump yarl to 1.9.11 (#125287)

* Bump yarl to 1.9.10

changelog: https://github.com/aio-libs/yarl/compare/v1.9.9...v1.9.10

* 11

bdraco committed:
Remove unneeded wrapping of URL in URL in network helper (#125265)

* Remove unneeded wrapping of URL in URL in network helper

* fix mocks

gjohansson-ST committed:
Code quality improvement on local_file (#125165)

* Code quality improvement on local_file

* Fix

* No translation

* Review comments

Kane610 committed:
Add filter run time for deCONZ air purifiers (#123306)

* Add filter run time for deCONZ air purifiers

* Add duration and second

* Fix review comments

* Update tests/components/deconz/snapshots/test_sensor.ambr

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Revert #122676 Yamaha discovery (#125216)

Revert Yamaha discovery

postlund committed:
Bump pyatv to 0.15.1 (#125412)

tl-sl committed:
Improve handling of old firmware versions (#125406)

* Update Info fixture with new fields from pysmlight 0.0.14

* Create repair if device is running unsupported firmware

* Add test for legacy firmware info

* Add strings for repair issue

balloob committed:
Migrate VoIP to use assist satellite (#125381)

* Migrate VoIP to assist satellite

* Fix flaky test

gjohansson-ST committed:
Remove deprecated aux_heat from ecobee (#125246)

AlexT59 committed:
Bump sfrbox-api to 0.0.10 (#125405)

* bump sfr_box requirement to 0.0.10

* upate manifest file

* Handle None values

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

dieselrabbit committed:
Remove support for area, device, or entity targets for screenlogic actions (#123432)

* Remove non-configentry service target

* Remove unneeded tests

* Remove unneeded issue strings

swistakm committed:
Bump blebox_uniapi to v2.5.0 (#124298)

blebox: bump blebox_uniapi to v2.5.0

balloob committed:
Migrate ESPHome to assist satellite (#125383)

* Migrate ESPHome to assist satellite

* Address comments

epenet committed:
Add model ID to sfr_box (#125400)

emontnemery committed:
Use PEP 695 for decorator typing with type aliases in zha (#124235)

mawoka-myblock committed:
Set min_power similar to max_power to support all inverters from apsystems (#124247)

Set min_power similar to max_power to support all inverters from apsystems ez1 series

tdfountain committed:
Improve consistency of sensor strings to reduce confusion in NUT (#124184)

Improve consistency of sensor strings to reduce confusion

tdfountain committed:
Move ambient sensors (temperature and humidity) to diagnostic in NUT (#124180)

Move ambient sensors (temperature and humidity) to Diagnostic

catsmanac committed:
Provide same entities for all Enphase_envoy CT types (#124531)

Provide same entities for all Enphase_envoy CT types.

swistakm committed:
Fix mired range in blebox color temp mode lights (#124258)

* fix: use default mired range in belbox lights running in color temp mode

* fix: ruff

emontnemery committed:
Fix UnboundLocalError in recorder (#125419)

catsmanac committed:
Improve coordinator test coverage for enphase_envoy (#122375)

* Improve coordinator test coverage for enphase_envoy

* rename to test_coordinator to test_init for enphase_envoy

* Mock pyenphase _obtain_token instead of httpx auth requests in enphase_envoy tests.

* Move EnvoyTokenAuth patch to mock_envoy of enphase_envoy

alengwenus committed:
Add tests for LCN climate and scene platform (#124466)

* Add tests for LCN climate and scene platform

* Add type hints

* Add snapshots for test_climate

* Add snapshots for test_scene

* Replace await_called assertion with snapshots

* Remove snapshots for simple status changes

* Test platform setup using snapshot_platform

* Fix type hints

* Patch homeassistant.components.lcn context instead of pypck module

* Fix side effects caused by patching PchkConnectionManager in lcn platform context

synesthesiam committed:
Use first media player announcement format for TTS (#125237)

* Use ANNOUNCEMENT format from first media player for tts

* Fix formatting

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

mj23000 committed:
Replace strings with constants in Bang & Olufsen testing (#125423)

Replace strings with constants in service calls

IceBotYT committed:
Nice G.O. code quality improvements (#124319)

* Bring Nice G.O. up to platinum

* Switch to listen in coordinator

* Tests

* Remove parallel updates from coordinator

* Unsub from events on config entry unload

* Detect WS disconnection

* Tests

* Fix tests

* Set unsub to None after unsubbing

* Wait 5 seconds before setting update error to prevent excessive errors

* Tweaks

* More tweaks

* Tweaks part 2

* Potential test for hass stopping

* Improve reconnect handling and test on Homeassistant stop event

* Move event handler to entry init

* Patch const instead of asyncio.sleep

---------

Co-authored-by: jbouwh <jan@jbsoft.nl>

mvn23 committed:
Add switch platform to opentherm_gw (#125410)

* WIP

* * Add switch platform
* Add tests for switches

* Remove unnecessary block_till_done-s

* Test that entities get added in a disabled state separately

* Convert to parametrized test

* Use fixture to add entities enabled.

hesselonline committed:
Add model ID to WallboxEntity (#125434)

* chore: Update WallboxEntity model ID to use CHARGER_PART_NUMBER_KEY

The WallboxEntity model ID is updated to use the CHARGER_PART_NUMBER_KEY value from the coordinator data. This change ensures consistency and accuracy in identifying the model of the Wallbox entity.

* Update WallboxEntity model ID to use CHARGER_PART_NUMBER_KEY

* chore: Update WallboxEntity model ID to use CHARGER_PART_NUMBER_KEY

* remove obsolete key from test

bdraco committed:
Bump yalexs to 8.6.4 (#125442)

adds a debounce to the updates to ensure we do
not request the activities api too often if
the websocket sends rapid updates

fixes #125277

bdraco committed:
Bump yarl to 1.10.0 (#125446)

changelog: https://github.com/aio-libs/yarl/compare/v1.9.11...v1.10.0

marcelveldt committed:
Add discovery schemas for Matter 1.3 power/energy sensors (#125403)

* Add missing discovery schemas for (Matter 1.3) Power/Energy measurements

* Prevent discovery of custom cluster if 1.3 cluster present

* add tests

* Use f-strings

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Gigatrappeur committed:
Change webhook id storage

Gigatrappeur commented:
I close this pull request and (finally), I separate my code in two pull request.  
The first concerned only the vacuum integration : #125457  
The second for webhook will be in after the merge of first pull request
